### ISSUE TITLE
Due date shouldn't be a required field when updating an invoice

### ISSUE THREAD

**Issue by ryanwilliams** [waynerobinson/xeroizer#395]
At the moment if you create an invoice you don't need to specify a due date. But if you wish to update an invoice you must specify a due date.

This doesn't seem right as Xero doesn't care if the due date is blank.

**Comment by rjaus** [comment]
Correct,  DueDate is only required for creation.  Update don't require the field to specified (although it should already exist).

Can you explain the use case in which this is a problem in practice? I would expect the object to already have a DueDate in any case where you're updating the invoice.

**Comment by ryanwilliams** [comment]
This is an issue because DueDate is optional on create. Xero doesn't require it. So you can create an invoice without one just fine, but then if you attempt to update it you can't unless you add a DueDate.

**Comment by rjaus** [comment]
Hey @ryanwilliams.  The issue is actually dependant on the <Status> of the invoice.  If invoice.status == DRAFT, no DueDate is required (on PUT or POST).  But if invoice.status == AUTHORISED DueDate is a required field (for PUT & POST).

If you want to enforce this behaviour in the wrapper, I'd change this line:

https://github.com/waynerobinson/xeroizer/blob/master/lib/xeroizer/models/invoice.rb#L93

From: 
```ruby
validates_presence_of :date, :due_date, :unless => :new_record?
```

To:
```ruby
validates_presence_of :date, :due_date, :if => :approved?
```

Just trying to think of the flow on effects before making this change in master.

**Comment by Samsinite** [comment]
Getting the same error over here, after a batch save all of the invoices report an error because `due_date` is blank even though it was saved on the xero side just fine.
